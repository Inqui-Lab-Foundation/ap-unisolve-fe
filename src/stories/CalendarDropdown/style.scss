.CalendarDropdownComp {
  .anticon-calendar {
    display: none;
  }
  .ant-picker {
    border-radius: 2.5rem;
    padding: 1.4rem 1.6rem;
    margin: 1rem;
    background: var(--color-white-3);
    border: 0.1rem solid var(--color-grey-light-1);
    box-sizing: border-box;
    border-radius: 6rem;
    color: var(--color-grey-light-3);
    font-size: 1.8rem;
  }
  .ant-picker-suffix > * {
    color: var(--color-grey-light-3);
  }
  .ant-picker-input {
    input {
      // text-align: center;
      color: var(--color-grey-light-3);
      // width: 5rem;
    }
  }
}
