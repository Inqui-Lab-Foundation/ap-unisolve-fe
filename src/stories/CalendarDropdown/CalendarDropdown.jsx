import React from "react";
import PropTypes from "prop-types";
import "./style.scss";
import { DownOutlined } from "@ant-design/icons";
// import moment from "moment";

//Antd Datepicker is used
import { DatePicker } from "antd";

export const CalendarDropdownComp = ({
    format,
    picker,
    placeholder,
    // options,
    value,
    // id,
    // ...props
}) => {
    const onChange = () => {};

    return (
        <div className="dropdown CalendarDropdownComp ">
            <DatePicker
                suffixIcon={<DownOutlined />}
                format={format}
                onChange={(e) => onChange(e)}
                picker={picker}
                placeholder={placeholder}
                id={"datePickerComp"}
                value={value}
                className="w-100"
            />
        </div>
    );
};

CalendarDropdownComp.propTypes = {
    onClick: PropTypes.func,
};

CalendarDropdownComp.defaultProps = {
    format: "",
    picker: "",
    placeholder: "",
    // value:"",
    // id:"datePickerComp"
};
