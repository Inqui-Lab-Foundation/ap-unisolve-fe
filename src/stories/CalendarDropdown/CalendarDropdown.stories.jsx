import React from 'react';
import moment from 'moment';

//Import Calendar component from CalendarDropdown.jsx
import { CalendarDropdownComp } from './CalendarDropdown';

export default {
    title: 'Example/CalendarDropdownComp',
    component: CalendarDropdownComp,
    argTypes: {
        format:"MMM",
        picker:"month",
        placeholder:"Month",
        value:moment(new Date()),
        id:"datePickerComp"

    },
};

const Template = (args) => <CalendarDropdownComp {...args} />;

//Calendar for Month slector
export const CalendarDropdownMonth = Template.bind({});
CalendarDropdownMonth.args = {
    format:"MMM",
    onChange:"undefined",
    picker:"month",
    placeholder:"Month",
    value:moment(new Date()),
    id:"datePickerComp"
 
};

//Calendar for Month slector
export const CalendarDropdownDay = Template.bind({});
CalendarDropdownDay.args = {
    format:"Do",
    picker:"",
    placeholder:"Day",
    value:moment(new Date()),
    id:"datePickerComp"
 
};

//Calendar for Day slector
export const CalendarDropdownYear = Template.bind({});
CalendarDropdownYear.args = {
    format:"YYYY",
    picker:"year",
    placeholder:"Year",
    value:moment(new Date()),
    id:"datePickerComp"
};

//Calendar for default slector
export const DefaultCalendar = Template.bind({});
DefaultCalendar.args = {
    format:"Do MMMM YYYY",
    // value:moment(new Date()),
    id:"datePickerComp"
};


