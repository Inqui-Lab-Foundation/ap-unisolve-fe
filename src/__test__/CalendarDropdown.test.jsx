import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {CalendarDropdownComp} from "../stories/CalendarDropdown/CalendarDropdown";
configure({ adapter: new Adapter() });
import moment from 'moment'

describe("MyComponent", () => {
    it('Create snapshot for date component', () => {  
        const props = {
            value: moment(new Date()),
            format: "MMM",
            picker: "month",
            placeholder: "Month",
            value:"20.01.2018",
            id:"datePickerComp"
        }
        const wrapper = shallow(<CalendarDropdownComp {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('check the onChange callback', () => {  
        const onChange = jest.fn()
        const props = {
            value: moment(new Date()),
            format: "MMM",
            picker: "month",
            placeholder: "Month",
            id:"datePickerComp"
        }
            const wrapper = shallow(<CalendarDropdownComp {...props}/>)
            wrapper.find('#datePickerComp').simulate('change', { target: {value: 'Hi'} });
            expect(wrapper.find('#datePickerComp').length).toBe(1);
        // expect(onChange).toHaveBeenCalledWith('Feb');
    });
});
