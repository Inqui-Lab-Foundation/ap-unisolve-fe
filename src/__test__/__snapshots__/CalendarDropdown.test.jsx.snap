// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Create snapshot for date component 1`] = `ShallowWrapper {}`;
